### ISSUE TITLE
[BUG] Unable to log artifact to Azure blob storage since artifact_uri pointing to local system

### ISSUE THREAD

**Issue by haizadtarik** [mlflow/mlflow#13470]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### Willingness to contribute

Yes. I can contribute a fix for this bug independently.

### MLflow version

- Client: 2.17.0
- Tracking server: 2.17.0

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 11
- **Python version**: 3.9.13

### Describe the problem

I'm running mlflow server and want to store artifacts in azure blob storage by running the CLI as follows:
```
mlflow server --default-artifact-root wasbs://<container>@<storage-account>.blob.core.windows.net/mlflowartifacts -h 0.0.0.0 -p 5000
```
But the artifacts is not uploaded to my blob storage and when i checked the `artifact_uri` using `mlflow.get_artifact_uri()` it is pointing to local file system (`file://<LOCAL-PATH>`) 
I pass connection string as environment variable named `AZURE_STORAGE_CONNECTION_STRING`

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
"""
Set environment variable and run mlflow server before running the script as follows:
set AZURE_STORAGE_CONNECTION_STRING=<AZURE-STORAGE-KEY>
mlflow server --default-artifact-root wasbs://<container>@<storage-account>.blob.core.windows.net/mlflowartifacts -h 0.0.0.0 -p 5000
"""

import mlflow
import os

print('Azure Key = ', os.environ.get("AZURE_STORAGE_CONNECTION_STRING"))

with mlflow.start_run():
    print(mlflow.get_artifact_uri())
```

### Stack trace

```
REPLACE_ME
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [X] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [X] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [X] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by WeichenXu123** [comment]
Thanks for reporting this ! Happy to see your fixing PR :)

**Comment by harupy** [comment]
@WeichenXu123

- Can we really reproduce the issue?
- What's the root cause?
- How to fix it?

**Comment by WeichenXu123** [comment]
oh sorry. Let me take a deeper look.

@haizadtarik 

For your reproducing code,
```
import mlflow
import os

print('Azure Key = ', os.environ.get("AZURE_STORAGE_CONNECTION_STRING"))

with mlflow.start_run():
    print(mlflow.get_artifact_uri())
```

before running `mlflow.get_artifact_uri()`,
did you set MLflow tracking uri like `mlflow.set_tracking_uri("https://127.0.0.1:5001")` ?

**Comment by haizadtarik** [comment]
@WeichenXu123 yes the tracking uri is set and correctly pointed to https://localhost:5000. Here is the output:
```
Artifact URI:  file:///D:/project/mlflow-artifacts/mlruns/0/b172a1562b16494e8bcf53cef32b5d12/artifacts
2024/10/18 16:45:29 INFO mlflow.tracking._tracking_service.client: 🏃 View run bouncy-shoat-368 at: http://localhost:5000/#/experiments/0/runs/b172a1562b16494e8bcf53cef32b5d12.
2024/10/18 16:45:29 INFO mlflow.tracking._tracking_service.client: 🧪 View experiment at: http://localhost:5000/#/experiments/0.
```

Btw I wrongly filled the bug issue, I need help to fix this as I cannot resolved this independently. 
Let me know if this is not reproducible on your side

**Comment by WeichenXu123** [comment]
could you try this command instead: @haizadtarik 

```
mlflow server \
    --host 0.0.0.0 \
    --port 5000 \
    --artifacts-destination wasbs://<container>@<storage-account>.blob.core.windows.net/mlflowartifacts \
    --serve-artifacts
```

**Comment by haizadtarik** [comment]
`mlflow.get_artifact_uri()` still pointing to local file system

**Comment by WeichenXu123** [comment]
I don't have a azure blob storage at hand, but I tested with other remote artifacts backend and reproduced similar issue.

@haizadtarik could you check the directory where you run `mlflow server ...` command, and delete `mlruns` directory under that directory ? and then run `mlflow server --default-artifact-root wasbs://<container>@<storage-account>.blob.core.windows.net/mlflowartifacts -h 0.0.0.0 -p 5000` again and test it.

**Comment by haizadtarik** [comment]
It works now!
You know why this happen?

**Comment by WeichenXu123** [comment]
`--default-artifact-root` configuration only works when MLflow creates backend storage the first time. 

When you have created backend storage (i.e., in your case, your local `mlruns` directory has been created), then the artifact root configuration is fixed.

**Comment by haizadtarik** [comment]
Got it now. Thank you for helping me on this.
Btw I think this need to be mentioned as note in the documentation. wdyt?

**Comment by khac** [comment]
```
--default-artifact-root configuration only works when MLflow creates backend storage the first time.
```
makes sense, ok closing this issue
